### PR TITLE
fix(localUsage): treat a non-positive scan window as empty, not unbounded

### PR DESCRIPTION
**Pull request by murdore**
## The bug

`readAllLocalUsage({ sinceDays: 0 })` read **every transcript on the machine**. Measured on release:

```
sinceDays: 0   ->  17,534 files, 35.9s
```

That's the widest possible scan in answer to the narrowest possible request, against a store that is 9.7 GB here. A negative value did the same.

The guard was `Number.isFinite(sinceDays) && sinceDays > 0`, which leaves the cutoff `undefined` for anything non-positive — disabling the filter entirely. The intent was that `Infinity` means "no filter"; `0` fell through the same door by accident.

## The fix

`Infinity` is now the only full-sweep signal, and the window is clamped at zero. Applied to all three readers, which shared the expression.

```
sinceDays: 0          ->  0 files,      1.9s
sinceDays: -5         ->  0 files,      0.5s
sinceDays: Infinity   ->  17,537 files  (full sweep intact)
```

**The CLI is unaffected and stays as documented.** `neurolink usage local --since 0` still means all history, because the command translates `0` to `Infinity` before it reaches a reader. That's a CLI convention rather than a reader one, and the two no longer have to agree by coincidence.

## Also here

`LocalUsageAggregateReport.failures` was documented as "CLIs that were registered but produced nothing". A failure is specifically a reader that could not be created, detected, or scanned — an absent store is reported under `notInstalled`. Conflating them misleads a caller about which of the two they're looking at.

## Proof

The new case asserts both directions. **Without the `Infinity` control, a reader that returned nothing for every window would satisfy the zero and negative assertions** — so the control is what makes the test mean anything.

Confirmed by restoring the old guard:

```
old guard   ✗ sinceDays 0 read transcripts — got 1 turns instead of 0
fixed       ✓ 16 passed
```

## Gates

```
tsc --noEmit       0 errors
pnpm run lint      0 errors
test:local-usage   16 passed, 0 failed
```

Found by CodeRabbit on #1482 after that PR had merged.

## Summary by CodeRabbit

* **New Features**
  * Added support for restricting local usage scans to a selected CLI.
  * Added validation for empty or unknown CLI filters and JSON output.

* **Bug Fixes**
  * Corrected local usage scanning for zero, negative, invalid, and infinite time-window values.
  * Applied consistent time-window behavior across supported local usage sources.

* **Documentation**
  * Clarified that usage report failures include readers that could not be created, detected, or scanned.

### CODE REVIEW

**Comment by Tara-ag** [comment]
🔒 CRITICAL: No security issues found - code review confirms no hardcoded secrets, injection vulnerabilities, or auth flaws.

⚠️ MAJOR: No blocking issues detected. This PR fixes a serious bug where `sinceDays: 0` would read ALL transcripts instead of nothing.

💡 MINOR: All changes are focused and well-documented. The three reader files use slightly different sentinel values (`undefined` vs `0`) for the "no filter" case, but both achieve the same semantic result correctly.

**Review Summary:**
- **Decision**: APPROVED
- **Files changed**: 6 (3 reader implementations + 1 type definition + 1 doc + 1 test)
- **Bug fixed**: Non-positive `sinceDays` values were incorrectly treated as unbounded windows
- **Test coverage**: New test added to prevent regression
- **Breaking changes**: None
- **Impact**: Self-contained to local usage module (~500 nodes affected per graph analysis)

The fix is minimal, correct, and properly tested. Ready to merge.

**Comment by Tara-ag** [comment]
✅ **YAMA REVIEW COMPLETE - APPROVED**

No issues found. The PR fixes a critical bug where `sinceDays: 0` or negative values would incorrectly read ALL transcripts instead of returning empty results.

**Verified:**
- ✅ All 3 reader implementations (claudeCodeReader, codexReader, openCodeReader) fixed correctly
- ✅ Documentation updated to reflect accurate failure semantics
- ✅ Test added to prevent regression
- ✅ No security vulnerabilities detected
- ✅ No breaking changes to public API
- ✅ Impact is self-contained (~500 nodes affected per graph analysis)

The fix follows the correct pattern: `Infinity` means "no time filter" (read everything), while non-positive values should return an empty window. Both sentinel approaches (`undefined` and `0`) achieve this semantically equivalent result.

Ready for merge.

**Review by Tara-ag — APPROVED** [review]

**Review by Tara-ag — APPROVED** [review]
## Summary

This PR fixes a critical performance regression where non-positive `sinceDays` values (0 or negative) would cause all local usage readers to scan every file on disk instead of returning an empty result.

**Impact:** On a real machine with 17,534 transcripts totaling 9.7 GB, `sinceDays: 0` was reading all files in 35.9 seconds - the widest possible scan in response to the narrowest possible request.

## Changes

### Bug Fix (3 files)
- `src/lib/localUsage/claudeCodeReader.ts`
- `src/lib/localUsage/codexReader.ts`  
- `src/lib/localUsage/openCodeReader.ts`

Changed the time filter logic from:
```typescript
Number.isFinite(sinceDays) && sinceDays > 0 ? Date.now() - ... : undefined
```

To:
```typescript
sinceDays === Infinity ? undefined : Date.now() - Math.max(0, sinceDays) * 86_400_000
```

This ensures only `Infinity` means "no filter", while zero/negative values produce a past cutoff date (effectively empty result).

### Documentation Clarification (2 files)
- `docs/api/type-aliases/LocalUsageAggregateReport.md`
- `src/lib/types/localUsage.ts`

Clarified that `failures` means "readers that could not be created, detected, or scanned" rather than "registered CLIs that produced nothing."

### Test Coverage (1 file)
- `test/continuous-test-suite-local-usage.ts`

Added comprehensive test validating:
- `sinceDays: 0` → 0 requests (empty window)
- `sinceDays: -5` → 0 requests (empty window)
- `sinceDays: Infinity` → full sweep (control case)

## Impact Analysis

- **Blast radius:** ~500 nodes impacted across 131 files
- **Affected flows:** `createClaudeCodeReader`, `createCodexReader`, `createOpenCodeReader`
- **Backward compatibility:** ✅ CLI convention preserved (`--since 0` still means all history via `Infinity` translation)
- **Breaking changes:** None
- **Performance improvement:** From 35.9s to 1.9s for zero-window queries on large machines

## Verification

- `tsc --noEmit`: 0 errors
- `pnpm run lint`: 0 errors
- New test passes: 16 passed, 0 failed

The fix has been confirmed by restoring the old guard which caused the test to fail (1 turn instead of 0).

## Decision

**APPROVED** - This is a well-targeted bug fix addressing a real performance issue with clear evidence, consistent implementation across all affected readers, proper documentation updates, and comprehensive test coverage. No findings submitted - the code meets all standards.

File: src/lib/localUsage/claudeCodeReader.ts [hunk 1]
```diff
@@ -219,11 +219,16 @@ export async function createClaudeCodeReader(): Promise<LocalUsageReader> {
...
+      // in answer to the narrowest possible request. Zero now means a
+      // zero-length window, which is what it reads as.
       const sinceDays = options?.sinceDays ?? DEFAULT_SINCE_DAYS;
       const cutoff =
-        Number.isFinite(sinceDays) && sinceDays > 0
-          ? Date.now() - sinceDays * 86_400_000
-          : undefined;
+        sinceDays === Infinity
+          ? undefined
+          : Date.now() - Math.max(0, sinceDays) * 86_400_000;
```
[the remark this answers is not in the document]
Comment (murdore) [reply src/lib/localUsage/claudeCodeReader.ts]: Good catch — my fix left a hole and this is it. Math.max(0, NaN) is NaN and every comparison against NaN is false, so sinceDays NaN passed the filter for every file: measured at 17,537 files in 32.8s, the exact unbounded sweep this PR exists to prevent, reached through a different door. The guard I replaced caught it with Number.isFinite and I dropped that check.

On the OpenCode reader it was worse than an over-read. The cutoff went into the SQL text and SQLite parses a bare NaN as an identifier — "no such column: NaN" — so the whole scan errored. That value is a bound parameter now; a number spliced into SQL text is still SQL text.

Both NaN and -Infinity are now asserted alongside 0 and -5, with the Infinity control keeping the case honest. Confirmed by dropping the guard: it fails with "a NaN sinceDays read transcripts instead of nothing".

**Comment by Tara-ag** [comment]
## Review Summary for PR #1496

### Decision: **APPROVED** ✅

This PR fixes a serious performance regression in local usage readers where invalid `sinceDays` values (0, negative, NaN) would cause unbounded scans reading all files instead of returning empty results.

### Changes Overview:

1. **Bug Fix in Three Readers** (`claudeCodeReader.ts`, `codexReader.ts`, `openCodeReader.ts`):
   - Added explicit NaN handling: `Number.isNaN(requestedDays) ? 0 : requestedDays`
   - Changed cutoff logic from `Number.isFinite(sinceDays) && sinceDays > 0` to proper bounds checking
   - Ensures only `Infinity` means "no time filter" (full sweep)
   - All other values produce appropriate time windows (including zero-length for ≤0)

2. **SQL Injection Prevention** (`openCodeReader.ts`):
   - Changed from string interpolation `${cutoffMs}` to bound parameter `?` with `.all(cutoffMs)`
   - Prevents SQL injection if somehow non-numeric values reach this code path

3. **Type Definition Updates** (`localUsage.ts`):
   - Updated type signatures to reflect new API (bound parameters for SQLite prepare)

4. **Documentation Updates**:
   - Clarified failure messages in API documentation

5. **Comprehensive Tests** (`test/continuous-test-suite-local-usage.ts`):
   - Added test case covering all edge cases: 0, -5, NaN, -Infinity, Infinity
   - Validates that non-positive values return empty results
   - Validates that Infinity still performs full sweep

### Impact Analysis:

Based on the code knowledge graph analysis:
- **Risk score**: medium (0.60)
- **Communities affected**: utils-constructor, test-tool, action-dist-constructor
- **Flows affected**: handleAuth, testExternalMCPConnection, fetchValidatedAccountUsage
- **Blast radius**: Localized to local usage scanning functionality; no impact on core SDK API

### Verification:

✅ Code changes are well-documented with extensive inline comments explaining the bug and fix
✅ Tests cover all edge cases including the problematic NaN case that previous fixes missed
✅ No breaking changes to public API
✅ Type definitions updated correctly
✅ SQL injection risk mitigated through bound parameters
✅ No hardcoded secrets or credentials
✅ Changes follow existing patterns in the codebase

### Conclusion:

This is a critical defensive fix for a performance/security issue that could have caused long-running scans consuming significant resources. The implementation is thorough, well-tested, and follows best practices. Ready to merge.

**Review by Tara-ag — APPROVED** [review]

**Review by Tara-ag — APPROVED** [review]
